web/packages/teleport/src/Discover/Discover.test.tsx

```diff
@@ -204,3 +208,91 @@ describe('location state', () => {
     ).not.toBeInTheDocument();
   });
 });
+
+const renderUpdate = (props: DiscoverUpdateProps) => {
+  const defaultPref = makeDefaultUserPreferences();
+  defaultPref.onboard.preferredResources = [Resource.WEB_APPLICATIONS];
+
+  mockUserContextProviderWith(
+    makeTestUserContext({ preferences: defaultPref })
+  );
+
+  const userAcl = getAcl();
+  const ctx = createTeleportContext({ customAcl: userAcl });
+
+  const MockComponent1 = () => {
+    const { agentMeta } = useDiscover();
+    return (
+      <>
+        {agentMeta.resourceName === 'saml2' ? agentMeta.resourceName : 'saml1'}
+      </>
+    );
+  };
+
+  const testViews: ResourceViewConfig[] = [
+    {
+      kind: ResourceKind.SamlApplication,
+      views() {
+        return [
+          {
+            title: 'MockComponent1',
+            component: MockComponent1,
+          },
+        ];
+      },
+    },
+  ];
+
+  return render(
+    <MemoryRouter
+      initialEntries={[
+        { pathname: cfg.routes.discover, state: { entity: '' } },
+      ]}
+    >
+      <TeleportContextProvider ctx={ctx}>
+        <DiscoverComponent eViewConfigs={testViews} updateFlow={props} />
+      </TeleportContextProvider>
+    </MemoryRouter>
+  );
+};
+
+test('update flow: renders single component based on resourceSpec', () => {
+  const resourceSpec: ResourceSpec = {
+    name: 'Connect My Computer',
+    kind: ResourceKind.ConnectMyComputer,
+    event: null,
+    icon: 'Laptop',
+    keywords: '',
+    hasAccess: true,
+  };
+
+  renderUpdate({ resourceSpec: resourceSpec, agentMeta: { resourceName: '' } });
+
+  expect(screen.queryAllByTestId(ResourceKind.Server).length).toBeFalsy();
```

Comment:
This seems to be easier to understand:
```suggestion
  expect(screen.queryByTestId(ResourceKind.Server)).not.toBeInTheDocument();
```
(same for other assertions)